File: 1/2021.04.06/153.Find Minimum in Rotated Sorted Array/java/solution.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0, right = nums.length - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] < nums[nums.length - 1]) {
                right = mid - 1;
            } else if (nums[mid] > nums[nums.length - 1]) {
                left = mid + 1;
            } else {
                right -= 1;
            }
        }
        return nums[left];
    }
}

File: 1/2021.04.06/153.Find Minimum in Rotated Sorted Array/python/solution.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        low, high = 0, len(nums) - 1
        while low <= high:
            mid = low + (high - low) // 2
            if nums[mid] > nums[-1]:
                low = mid + 1
            elif nums[mid] < nums[-1]:
                high = mid - 1
            else:
                high = mid - 1
        return nums[low]
